eclair-core/src/main/scala/fr/acinq/eclair/balance/ChannelsListener.scala

```diff
@@ -0,0 +1,66 @@
+package fr.acinq.eclair.balance
+
+import akka.Done
+import akka.actor.typed
+import akka.actor.typed.Behavior
+import akka.actor.typed.eventstream.EventStream
+import akka.actor.typed.scaladsl.adapter.ClassicActorRefOps
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import fr.acinq.bitcoin.ByteVector32
+import fr.acinq.eclair.balance.ChannelsListener.{ChannelData, ChannelDied, Command, GetChannels, GetChannelsResponse}
+import fr.acinq.eclair.channel.Helpers.Closing
+import fr.acinq.eclair.channel.{ChannelPersisted, ChannelRestored, HasCommitments}
+
+import scala.concurrent.Promise
+
+object ChannelsListener {
+
+  // @formatter:off
+  sealed trait Command
+  private final case class ChannelData(channelId: ByteVector32, channel: akka.actor.ActorRef, data: HasCommitments) extends Command
+  private final case class ChannelDied(channelId: ByteVector32) extends Command
+  final case class GetChannels(replyTo: typed.ActorRef[GetChannelsResponse]) extends Command
+  // @formatter:on
+
+  case class GetChannelsResponse(channels: Map[ByteVector32, HasCommitments])
+
+  def apply(ready: Promise[Done]): Behavior[Command] =
+    Behaviors.setup { context =>
+      context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[ChannelRestored](e => ChannelData(e.channelId, e.channel, e.data)))
+      context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[ChannelPersisted](e => ChannelData(e.channelId, e.channel, e.data)))
+      // since subscription is asynchronous, we send a fake event so we know when we are subscribed
+      context.system.eventStream ! EventStream.Publish(ChannelRestored(null, ByteVector32.Zeroes, null, null, null))
```

Comment:
This is dangerous, other actors who registered to this event will receive this fake event as well and it may break some of their internal logic...Unfortunately it looks like there's no built-in mechanism to check whether we've subscribed.

I'd prefer using a different `case class` than `ChannelRestored` or `ChannelPersisted` to check when we're subscribed. Can you define a new `case class CheckSubscribed(actor: ActorRef[Something])`, subscribe to this type of messages, send one where `actor = context.self` and wait to receive it through the event stream?

Reply:
Should we submit a PR to akka (cc @sstone)? This limitation is really annoying, not only here but in tests as well.

Reply:
Good idea, it's worth a shot and we'll learn a lot from their feedback.

Reply:
I tried a slightly more elegant, but still hackish version here: 0f933fe96 in the shorter term.